e2e/query-editor.spec.ts

```diff
@@ -18,165 +18,209 @@ test.describe('Prometheus query editor', () => {
     step
     type
     exemplars
-  `, async ({
-    readProvisionedDataSource,
-    explorePage,
-    page,
-  }) => {
-    const ds = await readProvisionedDataSource<DataSourcePluginOptionsEditorProps<PromOptions>>({ fileName: 'datasources.yml' });
-    
+  `, async ({ readProvisionedDataSource, explorePage, page }) => {
+    const ds = await readProvisionedDataSource<DataSourcePluginOptionsEditorProps<PromOptions>>({
+      fileName: 'datasources.yml',
+    });
+
     await explorePage.goto();
 
     await explorePage.datasource.set(ds.name);
     // query patterns
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.QueryBuilder.queryPatterns)).toBeVisible();
-    
+    await expect(explorePage.getByGrafanaSelector(selectors.components.QueryBuilder.queryPatterns)).toBeVisible();
+
     // explain
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.explain)).toBeVisible();
-      
+    await expect(
+      explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.explain)
+    ).toBeVisible();
+
     // editor toggle
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.editorToggle)).toBeVisible();
+    await expect(
+      explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.editorToggle)
+    ).toBeVisible();
 
     // options
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.options)).toBeVisible();
-    
+    await expect(
+      explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.options)
+    ).toBeVisible();
+
     // open the options
-    await explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.options)
-      .click();
+    await explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.options).click();
 
     // legend
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.legend)).toBeVisible();
-    
+    await expect(
+      explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.legend)
+    ).toBeVisible();
+
     // format
-    await expect(explorePage
-      .getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.format)).toBeVisible();
-    
-    // step
-    await expect(page.locator(`#${selectors.components.DataSource.Prometheus.queryEditor.step}`)).toBeVisible();
+    await expect(
+      explorePage.getByGrafanaSelector(selectors.components.DataSource.Prometheus.queryEditor.format)
+    ).toBeVisible();
+
+    // TODO: use selectors when  <EditorSwitch/> is fixed to pass data-testid
```

Comment:
is this component defined in the core prometheus package? if it has a label can that be used to select the switch instead?

this is non-blocking question

Reply:
Tried it, it didn't work; Im nor sure if it's a problem with EditorField components not passing the correct label or the plugin itself, will take a look. I opened two tickets for this, one in the new plugin-ui (since grafana/experimental is now deprecated)
https://github.com/grafana/plugin-ui/issues/114
https://github.com/grafana/grafana-amazonprometheus-datasource/issues/306